openapi_python_client/templates/model.pyi

```diff
@@ -105,27 +106,65 @@ class {{ model.reference.class_name }}:
             {{ construct(model.additional_properties, "prop_dict") | indent(12) }}
             additional_properties[prop_name] = {{ model.additional_properties.python_name }}
 
-        {{model.reference.module_name}}.additional_properties = additional_properties
+        {{model.reference.module_name}}._additional_properties = additional_properties
     {% else %}
-        {{model.reference.module_name}}.additional_properties = d
+        {{model.reference.module_name}}._additional_properties = d
     {% endif %}
 {% endif %}
         return {{model.reference.module_name}}
 
     {% if model.additional_properties %}
     @property
     def additional_keys(self) -> List[str]:
-        return list(self.additional_properties.keys())
+        return list(self._additional_properties.keys())
 
     def __getitem__(self, key: str) -> {{ additional_property_type }}:
-        return self.additional_properties[key]
+        return self._additional_properties[key]
 
     def __setitem__(self, key: str, value: {{ additional_property_type }}) -> None:
-        self.additional_properties[key] = value
+        self._additional_properties[key] = value
 
     def __delitem__(self, key: str) -> None:
-        del self.additional_properties[key]
+        del self._additional_properties[key]
 
     def __contains__(self, key: str) -> bool:
-        return key in self.additional_properties
+        return key in self._additional_properties
     {% endif %}
+
+{% macro required_property_type(prop) %}
+{%- if prop.nullable -%}
+Optional[{{prop.get_base_type_string()}}]
+{%- else -%}
+{{prop.get_base_type_string()}}
+{%- endif -%}
+{% endmacro %}
+
+    {% if model.additional_properties %}
+    def get(self, key, default=None) -> Optional[{{ additional_property_type }}]:
+        return self._additional_properties.get(key, default)
+    {% endif %}
+
+    {% for property in model.required_properties + model.optional_properties %}
+        {% set backing_member = "self._" + property.python_name %}
+        {% if property.required %}
+    @property
+    def {{property.python_name}}(self) -> {{required_property_type(property)}}:
+        return {{backing_member}}
+        {% else %}
+    @property
+    def {{property.python_name}}(self) -> {{required_property_type(property)}}:
+        if isinstance(self._{{property.python_name}}, Unset):
+            raise NotPresentError(self, "{{property.python_name}}")
+        return {{backing_member}}
+        {% endif %}
+
+    @{{property.python_name}}.setter
+    def {{property.python_name}}(self, value:{{required_property_type(property)}}) -> None:
+        {{backing_member}} = value
+
+        {% if not property.required %}
+    @{{property.python_name}}.deleter
+    def {{property.python_name}}(self) -> None:
+        {{backing_member}} = UNSET
+        {% endif %}
+    {% endfor %}
```

Comment:
This ended up being required since mypy (probably correctly) didn't recognize the different type hints between the setter and getter, but I think this is probably the more correct implementation anyway.